### PR TITLE
Flyout design guidelines

### PR DESCRIPTION
**Pull request by Blackbaud-DeniseCisneros**

### CODE REVIEW

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-flyout?_sp.spa=pr-flyout-design-guidelines](https://developer.blackbaud.com/skyux-flyout?_sp.spa=pr-flyout-design-guidelines)

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-flyout?_sp.spa=pr-flyout-design-guidelines](https://developer.blackbaud.com/skyux-flyout?_sp.spa=pr-flyout-design-guidelines)

File: src/app/docs/flyout/flyout-docs.component.html [hunk 1]
```diff
@@ -65,4 +66,395 @@
...
+
+        <sky-docs-design-guideline-thumbnail
+          imageSource="~/assets/img/guidelines/flyout/opt-list.png"
+          showBorder="true"
+        >
+        </sky-docs-design-guideline-thumbnail>
+      </sky-docs-design-guideline>
+
+      <sky-docs-design-guideline
+        heading="Mobile behavior"
```
Comment (blackbaud-johnly) [inline src/app/docs/flyout/flyout-docs.component.html:432]:
```suggestion
        heading="Responsiveness"
```
Also, I think we should move this section to be a subsection in the Behavior and states section. I searched for existing Design guidelines with Responsiveness sections, and all the ones I found were in the Behavior and states section, not the Options section.

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-flyout?_sp.spa=pr-flyout-design-guidelines](https://developer.blackbaud.com/skyux-flyout?_sp.spa=pr-flyout-design-guidelines)

File: src/app/docs/flyout/flyout-docs.component.html [hunk 2]
```diff
@@ -65,4 +66,395 @@
       sourceCodePath="src/app/public/plugin-resources/code-examples/flyout/grid"
     ></sky-docs-code-example>
   </sky-docs-code-examples>
+
+  <sky-docs-design-guidelines>
+    <sky-docs-demo-page-section
```
[the remark this answers is not in the document]
Comment (Blackbaud-DeniseCisneros) [reply src/app/docs/flyout/flyout-docs.component.html:71]: Moved to bottom right before Related information. Done

File: src/app/docs/flyout/flyout-docs.component.html [hunk 3]
```diff
@@ -65,4 +66,395 @@
...
+        heading="Use when"
+        headingStyle="success"
+      >
+      <p class="sky-docs-text-block">
+        A flyout is a container that slides out from the side of the page and lays over page content to display large amounts of supplementary information related to user tasks. A flyout can contain lists, record views, graphs, or other information displays.
+      </p>
+
+      <sky-docs-design-guideline-thumbnail
+        imageSource="~/assets/img/guidelines/flyout/flyout-skeleton.png"
+        showBorder="true"
```
[the remark this answers is not in the document]
Comment (Blackbaud-DeniseCisneros) [reply src/app/docs/flyout/flyout-docs.component.html:84]: Done

File: src/app/docs/flyout/flyout-docs.component.html [hunk 4]
```diff
@@ -65,4 +66,395 @@
...
+      </sky-docs-design-guideline>
+
+      <sky-docs-design-guideline>
+        <p class="sky-docs-text-block">
+          Use a flyout to open records in a list so that users can return to the list view after completing quick tasks on the record pages. If users require a full-screen view of details or need to take multiple actions on the record, navigate to a full page instead.
+        </p>
+
+        <sky-docs-design-guideline-thumbnail
+          imageSource="~/assets/img/guidelines/flyout/flyout-list-view.png"
+          showBorder="true"
```
[the remark this answers is not in the document]
Comment (Blackbaud-DeniseCisneros) [reply src/app/docs/flyout/flyout-docs.component.html:98]: Done

File: src/app/docs/flyout/flyout-docs.component.html [hunk 5]
```diff
@@ -65,4 +66,395 @@
...
+      </sky-docs-design-guideline>
+
+      <sky-docs-design-guideline>
+        <p class="sky-docs-text-block">
+          Use a flyout to drill into an analytical insight so that users can quickly glance into what makes up that insight without spawning another workflow.
+        </p>
+
+        <sky-docs-design-guideline-thumbnail
+          imageSource="~/assets/img/guidelines/flyout/flyout-dashboard-view.png"
+          showBorder="true"
```
[the remark this answers is not in the document]
Comment (Blackbaud-DeniseCisneros) [reply src/app/docs/flyout/flyout-docs.component.html:112]: Done

File: src/app/docs/flyout/flyout-docs.component.html [hunk 6]
```diff
@@ -65,4 +66,395 @@
...
+      </sky-docs-design-guideline>
+
+      <sky-docs-design-guideline>
+        <p class="sky-docs-text-block">
+          Use a flyout to drill into child records so that users can retain the context of the parent record. The flyout only covers part of the current page, so users can refer back to it for context.
+        </p>
+
+        <sky-docs-design-guideline-thumbnail
+          imageSource="~/assets/img/guidelines/flyout/flyout-record-view.png"
+          showBorder="true"
```
[the remark this answers is not in the document]
Comment (Blackbaud-DeniseCisneros) [reply src/app/docs/flyout/flyout-docs.component.html:126]: Done

File: src/app/docs/flyout/flyout-docs.component.html [hunk 7]
```diff
@@ -65,4 +66,395 @@
...
+        headingStyle="danger"
+      >
+        <p class="sky-docs-text-block">
+          Do not use a flyout to navigate from one record to another instance of the same record type. For example, if users on a constituent record select a link to a different constituent record, navigate to the new record. Do not open it in a flyout.
+        </p>
+
+        <sky-docs-design-guideline-thumbnail
+          imageSource="~/assets/img/guidelines/flyout/not-record-to-record.png"
+          showBorder="true"
+        >
```
[the remark this answers is not in the document]
Comment (Blackbaud-DeniseCisneros) [reply src/app/docs/flyout/flyout-docs.component.html:146]: Done

File: src/app/docs/flyout/flyout-docs.component.html [hunk 8]
```diff
@@ -65,4 +66,395 @@
...
+      </sky-docs-design-guideline>
+
+      <sky-docs-design-guideline>
+        <p class="sky-docs-text-block">
+          Do not use a flyout to navigate from a child page to its parent page. For example, if users on a gift record select a link to return to a constituent record, navigate to a full-page view of the constituent record. Do not open it in a flyout.
+        </p>
+
+        <sky-docs-design-guideline-thumbnail
+          imageSource="~/assets/img/guidelines/flyout/not-child-to-parent.png"
+          showBorder="true"
```
[the remark this answers is not in the document]
Comment (Blackbaud-DeniseCisneros) [reply src/app/docs/flyout/flyout-docs.component.html:157]: Done

File: src/app/docs/flyout/flyout-docs.component.html [hunk 9]
```diff
@@ -65,4 +66,395 @@
...
+      </sky-docs-design-guideline>
+
+      <sky-docs-design-guideline>
+        <p class="sky-docs-text-block">
+          Do not use a flyout when another flyout is already open. If users select a link in a flyout, close the flyout and navigate to the new link location. In other words, limit flyouts to a single drill-in. On further drilling, open a full page.
+        </p>
+
+        <sky-docs-design-guideline-thumbnail
+          imageSource="~/assets/img/guidelines/flyout/not-double-panel.png"
+          showBorder="true"
```
[the remark this answers is not in the document]
Comment (Blackbaud-DeniseCisneros) [reply src/app/docs/flyout/flyout-docs.component.html:171]: Done

File: src/app/docs/flyout/flyout-docs.component.html [hunk 10]
```diff
@@ -65,4 +66,395 @@
...
+      </sky-docs-design-guideline>
+
+      <sky-docs-design-guideline>
+        <p class="sky-docs-text-block">
+          Do not use a flyout on pages that don't make sense when it closes. For example, on a page to enter grades for a class of students, displaying the names of students doesn't provide value without the grade entry form, so don't use a flyout to display the grade entry form.
+        </p>
+
+        <sky-docs-design-guideline-thumbnail
+          imageSource="~/assets/img/guidelines/flyout/not-need-details.png"
+          showBorder="true"
```
[the remark this answers is not in the document]
Comment (Blackbaud-DeniseCisneros) [reply src/app/docs/flyout/flyout-docs.component.html:185]: Done

File: src/app/docs/flyout/flyout-docs.component.html [hunk 11]
```diff
@@ -65,4 +66,395 @@
...
+      </sky-docs-design-guideline>
+
+      <sky-docs-design-guideline>
+        <p class="sky-docs-text-block">
+          Do not use a flyout to display a small amount of additional information. Flyouts are for large amounts of supplementary information. To display smaller amounts of information, use <a href="https://developer.blackbaud.com/skyux-popovers/docs/popover">a popover component</a> or <a href="https://developer.blackbaud.com/skyux-lists/docs/repeater">an expandable repeater</a>.
+        </p>
+
+        <sky-docs-design-guideline-thumbnail
+          imageSource="~/assets/img/guidelines/flyout/not-small-content.png"
+          showBorder="true"
```
[the remark this answers is not in the document]
Comment (Blackbaud-DeniseCisneros) [reply src/app/docs/flyout/flyout-docs.component.html:199]: Done

File: src/app/docs/flyout/flyout-docs.component.html [hunk 12]
```diff
@@ -65,4 +66,395 @@
...
+      </sky-docs-design-guideline>
+
+      <sky-docs-design-guideline>
+        <p class="sky-docs-text-block">
+          Do not use a flyout to display items that need a long time to load such as system-generated reports. Instead, let users continue to use the system, and use a <a href="https://developer.blackbaud.com/skyux-toast/docs/toast">toast</a> to let them know when the content is available.
+        </p>
+
+        <sky-docs-design-guideline-thumbnail
+          imageSource="~/assets/img/guidelines/flyout/not-long-load.png"
+          showBorder="true"
```
[the remark this answers is not in the document]
Comment (Blackbaud-DeniseCisneros) [reply src/app/docs/flyout/flyout-docs.component.html:213]: Done

File: src/app/docs/flyout/flyout-docs.component.html [hunk 13]
```diff
@@ -65,4 +66,395 @@
...
+      </sky-docs-design-guideline>
+
+      <sky-docs-design-guideline>
+        <p class="sky-docs-text-block">
+          Do not use a flyout to link to external sites. Open external sites in new tabs. For example, when users select a constituent address, a Google map of that address opens in a new tab.
+        </p>
+
+        <sky-docs-design-guideline-thumbnail
+          imageSource="~/assets/img/guidelines/flyout/not-external.png"
+          showBorder="true"
```
[the remark this answers is not in the document]
Comment (Blackbaud-DeniseCisneros) [reply src/app/docs/flyout/flyout-docs.component.html:227]: Done

File: src/app/docs/flyout/flyout-docs.component.html [hunk 14]
```diff
@@ -65,4 +66,395 @@
...
+      </sky-docs-design-guideline>
+
+      <sky-docs-design-guideline>
+        <p class="sky-docs-text-block">
+          Do not use a flyout for tasks where users require a full-screen view of details or need to take multiple actions within the view. Navigate to a full page instead.
+        </p>
+
+        <sky-docs-design-guideline-thumbnail
+          imageSource="~/assets/img/guidelines/flyout/not-need-full.png"
+          showBorder="true"
```
[the remark this answers is not in the document]
Comment (Blackbaud-DeniseCisneros) [reply src/app/docs/flyout/flyout-docs.component.html:241]: Done

File: src/app/docs/flyout/flyout-docs.component.html [hunk 15]
```diff
@@ -65,4 +66,395 @@
...
+              </sky-docs-thumbnail>
+            </td>
+            <td class="sky-docs-table-cell">
+              <sky-docs-thumbnail
+                imageSource="~/assets/img/guidelines/flyout/ixn-bg-active-3.png"
+                showBorder="true"
+              >
+              </sky-docs-thumbnail>
+            </td>
+          </tr>
```
[the remark this answers is not in the document]
Comment (Blackbaud-DeniseCisneros) [reply src/app/docs/flyout/flyout-docs.component.html:285]: It's the table styles

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-flyout?_sp.spa=pr-flyout-design-guidelines](https://developer.blackbaud.com/skyux-flyout?_sp.spa=pr-flyout-design-guidelines)

File: src/app/docs/flyout/flyout-docs.component.html [hunk 16]
```diff
@@ -65,4 +66,395 @@
...
+        <p class="sky-docs-text-block">
+          When users select a non-interactive element on the background page, the flyout closes. However, when users select a non-interactive element on the background page on touch devices, the flyout does not close because users need to be able touch the background page in order to scroll.
+        </p>
+
+        <table class="sky-docs-table">
+          <tr>
+            <td class="sky-docs-table-cell">
+              <sky-docs-thumbnail
+                imageSource="~/assets/img/guidelines/flyout/ixn-bg-active-1.png"
+                showBorder="true"
```
[the remark this answers is not in the document]
Comment (blackbaud-johnly) [reply src/app/docs/flyout/flyout-docs.component.html:267]: Just realized I meant to ping @Blackbaud-ToddRoberts to see if he thinks we should add some sort of caption here.

File: src/app/docs/flyout/flyout-docs.component.html [hunk 17]
```diff
@@ -65,4 +66,395 @@
...
+              >
+              </sky-docs-thumbnail>
+            </td>
+          </tr>
+        </table>
+      </sky-docs-design-guideline>
+
+      <sky-docs-design-guideline
+        heading="Help panel"
```
[the remark this answers is not in the document]
Comment (blackbaud-johnly) [reply src/app/docs/flyout/flyout-docs.component.html:358]: I pinged Colby to ask him what he thinks we need to do after we transition away from Help Widget.

File: src/app/docs/flyout/flyout-docs.component.html (hunk 1 above)
Comment (blackbaud-johnly) [reply src/app/docs/flyout/flyout-docs.component.html]: Just following up to see if you decided against the second suggestion. If you think this particular *Responsiveness* section should be under *Options*, that works. But all the other *Responsiveness* sections I found were under *Behavior and states*.

**Review by blackbaud-johnly — APPROVED** [review]

File: src/app/docs/flyout/flyout-docs.component.html (hunk 1 above)
Comment (Blackbaud-DeniseCisneros) [reply src/app/docs/flyout/flyout-docs.component.html]: I made the heading change but overlooked moving the section. That's done now.

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-flyout?_sp.spa=pr-flyout-design-guidelines](https://developer.blackbaud.com/skyux-flyout?_sp.spa=pr-flyout-design-guidelines)

**Review by blackbaud-johnly — APPROVED** [review]
Re-approving, but it looks like one of the checks is failing. Let me know if you need help with that.

File: src/app/docs/flyout/flyout-docs.component.html (hunk 16 above)
[the remark this answers is not in the document]
Comment (Blackbaud-ToddRoberts) [reply src/app/docs/flyout/flyout-docs.component.html:267]: honestly a lot of the examples in this section are not that useful - they would be better served as animations. if you want I can go through and remove the ones I think aren't doing us any good.

**Comment by Blackbaud-AlexKingman** [comment]
Looks like a random timeout. Forced a re-build...lets see if that fixes it.

File: src/app/docs/flyout/flyout-docs.component.html (hunk 16 above)
[the remark this answers is not in the document]
Comment (Blackbaud-DeniseCisneros) [reply src/app/docs/flyout/flyout-docs.component.html:267]: If y'all don't mind, let's address the unhelpful examples in a new PR so we can get this one out the door.

**Pull request merged**